src/cljc/rems/common/application_util.cljc

```diff
@@ -27,3 +29,25 @@
 
 (defn is-handler? [application user]
   (contains? (workflow-handlers application) user))
+
+(defn- str-to-int [s]
+  #?(:clj (Integer/parseInt s)
+     :cljs (js/parseInt s 10)))
+
+(defn- printToConsole [s]
+  #?(:clj (println s)
+     :cljs (js/console.log s)))
+
+(defn parse-sortable-external-id [external-id]
+  ;; the idea is to parse all numbers from the string, then add them up to form an external id - 
+  ;; comes from a theory that it is somehow tied to the date or the numbers are created in sequence
+  (cond
+    (if (seq (re-seq #"\d+" external-id)) "true" "false") (reduce + (map (fn [str] (str-to-int str)) (re-seq #"\d+" external-id)))
+    :else external-id))
+
+;; put test here
+(deftest test-parse-sortable-external-id
+  (is (= 2030 (parse-sortable-external-id "2020/10")))
+  (is (= 2092 (parse-sortable-external-id "THL/54/14.01.00/2002-rems/21")))
```

Comment:
This should probably be `["THL/" 54 "/14.01.00/" 2002 "-rems/" 21]`

And THL/2492/4.00.00/2020 would be `["THL/" 2492 "/4.00.00/" 2020]`

Reply:
Why not
```
["THL/" 54 "/" 14 "." 1 "." 0 "/" 2002 "-rems/" 21]
["THL/" 2492 "/" 4 "." 0 "." 0 "/" 2020]
```
more consistent handling of numbers and results in the same ordering (assuming 0-padding is consistent)

Reply:
I'm fine with any if there are clear cases that show what the behavior is. It's a bit difficult to say if how some of these number-like parts should be sorted, and if they only ever have numbers.

Reply:
It's quite likely that we have to amend the cases later.

Reply:
One of the possible complications is that we have to detect which part is a year and sort by that first.